keps/sig-node/4580-deprecate-kubelet-runonce/README.md

```diff
@@ -0,0 +1,348 @@
+checklist items _must_ be updated for the enhancement to be released.
+-->
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Deprecated and removed kubelet support for RunOnce mode.
+
+## Motivation
+
+* Runonce does not work in systemd mode.
+* Runonce mode doesn't support many newer pod features (init containers)
+* The pod lifecycle for runonce mode is even less well defined than normal.
+* The mode appears to be entirely untested in CI, except for a few unit tests.
+
+### Goals
+
+* Mark the `RunOnce` field in `KubeletConfiguration` and the `runonce` flag of kubelet as deprecated.
+* Remove kubelet support for RunOnce mode.
+
+### Non-Goals
+
+Runonce mode will be removed immediately. The deprecation and removal process will be gradual and feature gate to increase awareness among potential users.
+
+## Proposal
+
+This proposal outlines plans to deprecate and remove runonce mode in kubelet.
+
+### Risks and Mitigations
+
+Some people still rely on this feature, but podman addresses the same use case with more well-supported way, ref: https://docs.podman.io/en/latest/markdown/podman-kube.1.html. Affected users can migrate to *podman kube subcommand* on demand.
```

Comment:
```suggestion
Some people may still rely on this feature, but podman addresses the same use case with more well-supported way, ref: https://docs.podman.io/en/latest/markdown/podman-kube.1.html. Affected users can migrate to *podman kube subcommand* on demand.
```

I don't think we actually know of any users (unless you do?). Maybe someone will come forth during the review process. Otherwise, that's the purpose of the feature-gated deprecation rollout.